warrior/Framework/ClassUtils/netconf.py

```diff
@@ -348,18 +348,24 @@ def __receive(self):
                         self.__wait_resp.set()
                         self.close()
                         return False
+
                 if len(self.__wait_string) != 0 and self.__wait_string[0]:
                     waitstr = self.__wait_string
                     for notification in self.__notification_list:
                         pNote("Checking notification: "
                               "##{}##".format(notification))
                         match = False
                         xml = etree.fromstring(notification)
-                        temp = xml.xpath(waitstr[0], namespaces=waitstr[1])
-                        if isinstance(temp, bool) and temp:
-                            match = True
-                        elif isinstance(temp, list) and len(temp) > 0:
-                            match = True
+                        #Contains list of xpath
+                        xpath_list = waitstr[0].split(",")
+                        for xpath in xpath_list:
+                            #Validation for the xpath given
+                            status = xml.xpath(xpath, namespaces=waitstr[1])
+                            if status:
+                              match = True
+                            else:
+                              match = False
+                              break
```

Comment:
Code block can be better written as:
```
match, xpath_list = True, waitstr[0].split(",")
    for xpath in xpath_list:
        #Validation for the xpath given
        if not xml.xpath(xpath, namespaces=waitstr[1]):
            match = False
            break
```

Reply:
Hi @sanikakulkarni,

Made necessary changes and committed.